### ISSUE TITLE
Modular Testing with Apollo

### ISSUE THREAD

**Issue by GrayHHHH** [lgsvl/simulator#1276]
Hi guys, I have tried Modular Testing which uses Ground Truth in Apollo, but I assume that ego vehicle only receives all obstacles information within MaxDistance, but it doesn't consider the occlusion situation, i.e. not receive information of obstacles which ego can't see because of occlusion caused by other vehicle. Is my understanding correctly? If it is, is there any way to improve it, in order to make it more realistic?

Thank you so much for your reply!

**Comment by EricBoiseLGSVL** [comment]
@GrayHHHH yes you are correct.  You could make a new ground truth sensor that raycasts to the mesh to see if the obstacle is visible.  You can also use renderer.isvisible but it is not very accurate.  We will have an example of raycast method in our upcoming release with npcs to see if they are visible to the active ego vehicle.  Hope to release this very soon.
